packages/sdk/src/modules/JsonRpcDatasource.ts

```diff
@@ -0,0 +1,235 @@
+  constructor({ network }: JsonRpcDatasourceOptions) {
+    super({ network });
+  }
+
+  async getBalance({ address }: GetBalanceOptions) {
+    if (!address) {
+      throw new OrditSDKError("Invalid request");
+    }
+
+    return rpc[this.network].call<number>(
+      "Address.GetBalance",
+      { address },
+      rpc.id,
+    );
+  }
+
+  async getInscription({
+    id: _id,
+    decodeMetadata = false,
+  }: GetInscriptionOptions) {
+    if (!_id) {
+      throw new OrditSDKError("Invalid request");
+    }
+
+    const id = outpointToIdFormat(_id);
+
+    const inscription = await rpc[this.network].call<Inscription>(
+      "Ordinals.GetInscription",
+      { id },
+      rpc.id,
+    );
+
+    return DatasourceUtility.parseInscription(inscription, { decodeMetadata });
+  }
+
+  async getInscriptionUTXO({ id: _id }: GetInscriptionUTXOOptions) {
+    if (!_id) {
+      throw new OrditSDKError("Invalid request");
+    }
+
+    const id = outpointToIdFormat(_id);
+
+    return rpc[this.network].call<UTXO>(
+      "Ordinals.GetInscriptionUtxo",
+      { id },
+      rpc.id,
+    );
+  }
+
+  async getInscriptions({
+    creator,
+    owner,
+    mimeType,
+    mimeSubType,
+    outpoint,
+    decodeMetadata = false,
+    sort = "asc",
+    limit = 25,
+    next: _next = null,
+  }: GetInscriptionsOptions) {
+    let inscriptions: Inscription[] = [];
+    let next = _next;
+    do {
+      // eslint-disable-next-line no-await-in-loop
+      const { inscriptions: _inscriptions, pagination } = await rpc[
+        this.network
+      ].call<OrdinalsGetInscriptionsJsonRpcResponse>(
+        "Ordinals.GetInscriptions",
+        {
+          filter: { creator, owner, mimeType, mimeSubType, outpoint },
+          sort: { number: sort },
+          pagination: { limit, next },
+        },
+        rpc.id,
+      );
+      inscriptions = inscriptions.concat(_inscriptions);
+      next = pagination.next;
+    } while (next !== null);
+    return DatasourceUtility.parseInscriptions(inscriptions, {
+      decodeMetadata,
+    });
+  }
+
+  async getSpendables({
+    address,
+    value,
+    rarity = ["common"],
+    filter = [],
+    limit = 200,
+    type = "spendable",
+  }: GetSpendablesOptions) {
+    if (!address || Number.isNaN(value) || !value) {
+      throw new OrditSDKError("Invalid request");
+    }
+
+    return rpc[this.network].call<UTXOLimited[]>(
+      "Address.GetSpendables",
+      {
+        address,
+        value,
+        safetospend: type === "spendable",
+        allowedrarity: rarity,
+        filter,
+        limit,
+      },
+      rpc.id,
+    );
+  }
+
+  async getTransaction({
+    txId,
+    ordinals = true,
+    hex = false,
+    witness = true,
+    decodeMetadata = true,
+  }: GetTxOptions) {
+    if (!txId) {
+      throw new OrditSDKError("Invalid request");
+    }
+
+    const tx = await rpc[this.network].call<Transaction>(
+      "Transactions.GetTransaction",
```

Comment:
Can you expose an enum that encapsulates all these RPC methods? I think it's being used on Ordzaar as well. So it's easier to update

Reply:
ah nvm, we use the endpoints